src/components/PrivateRoute.jsx

```diff
@@ -10,22 +10,27 @@ import { checkIfValidDomain } from '../../backend/authBackend';
 const PrivateRoute = (props) => {
   const { component: Component, ...rest } = props;
   const { authenticated } = useAuth();
-  const [email, setEmail] = useState(false);
+  const [email, setEmail] = useState(undefined);
 
   useEffect(() => {
+    let relevant = true;
     const fetchData = async () => {
       try {
         const value = await checkIfValidDomain();
-        setEmail(value);
-
+        if (relevant) {
+          setEmail(value);
+        }
       } catch (err) {
-        console.error(err);
+        setEmail(false)
```

Comment:
This is overly broad and will make (possibly unrelated) failures impossible to observe.  *Any* error will be swallowed and lost entirely here.
Please keep on logging the error.

Reply:
Will add back! Thanks for the feedback